hashprobe/hashprobe.go

```diff
@@ -129,145 +129,101 @@ func (t *Uint32Table) Probe(keys []uint32, values []int32) int {
 //
 // The table uses the following memory layout:
 //
-//		[bitmap][keys][values]
+//		[group 0][group 1][...][group N]
 //
-// - The bitmap is used to determine which slots of the table are occupied,
-//   when a bit is set to 1, the key and value at the corresponding index in
-//   the arrays of keys and values have been assigned.
+// Each group contains up to 7 key/value pairs, and is exactly 64 bytes in size,
```

Comment:
Interesting. How does 7 divide into 64? Is there 8 bytes of overhead?

Reply:
There is padding to reach 64 bytes indeed, otherwise some groups may span two CPU cache lines and would be twice the cost to load into memory.